evpp/sockets.cc

```diff
@@ -20,28 +20,11 @@ std::string strerror(int e) {
         LocalFree(buf);
         return s;
     }
-
-#elif defined(H_OS_MACOSX)
-    char buf[2048] = {};
-    int rc = strerror_r(e, buf, sizeof(buf) - 1); // XSI-compliant
-    if (rc == 0) {
-        return std::string(buf);
-    }
 #else
     char buf[2048] = {};
-    #if (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && ! _GNU_SOURCE
-    int rc = strerror_r(e, buf, sizeof(buf) - 1); // XSI-compliant
-    if (rc == 0) {
-        return std::string(buf);
-    }
-    #else
-    const char* s = strerror_r(e, buf, sizeof(buf) - 1); // GNU-specific
-    if (s) {
-        return std::string(s);
-    }
-    #endif
+    strerror_r(e, buf, sizeof(buf) - 1);
```

Comment:
This cannot work on MACOSX

Reply:
True. It won't work with all GNU versions because buf is not guaranteed to be filled. I think OSX would be fine. I'll drop this commit from the set.